internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot/efivars.go

```diff
@@ -81,5 +74,99 @@ func WriteVariable(c efivario.Context, name, value string) error {
+//
+//nolint:gocyclo
+func CreateBootEntry(installDisk, sdBootFilePath string) error {
+	// mount EFI vars as rw
+	if err := unix.Mount("efivarfs", constants.EFIVarsMountPoint, "efivarfs", unix.MS_REMOUNT, ""); err != nil {
+		return err
+	}
+
+	defer unix.Mount("efivarfs", constants.EFIVarsMountPoint, "efivarfs", unix.MS_REMOUNT|unix.MS_RDONLY, "") //nolint:errcheck
+
+	rawBootOrderData, _, err := efivarfs.Read(efivarfs.ScopeGlobal, "BootOrder")
+	if err != nil {
+		return fmt.Errorf("failed to read BootOrder: %w", err)
+	}
+
+	bootOrder, err := efivarfs.UnmarshalBootOrder(rawBootOrderData)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal BootOrder: %w", err)
+	}
+
+	// Let's assume we are adding a new boot entry for Talos Linux UKI.
+	talosBootIndex := len(bootOrder)
+
+	for _, idx := range bootOrder {
+		bootEntry, err := efivarfs.GetBootEntry(int(idx))
+		if err != nil {
+			return fmt.Errorf("failed to get boot entry %d: %w", idx, err)
+		}
+
+		if bootEntry.Description == TalosBootEntryDescription {
+			// If we already have a Talos Linux UKI boot entry, we will use its index.
+			// This allows us to update the existing entry instead of creating a new one.
+			talosBootIndex = int(idx)
+
+			break
+		}
+	}
+
+	info, err := blkid.ProbePath(installDisk, blkid.WithSkipLocking(true))
+	if err != nil {
+		return fmt.Errorf("failed to probe install disk %q: %w", installDisk, err)
+	}
+
+	efiPartInfo := xslices.Filter(info.Parts, func(part blkid.NestedProbeResult) bool {
+		return part.PartitionLabel != nil && *part.PartitionLabel == constants.EFIPartitionLabel
+	})
+
+	if len(efiPartInfo) == 0 {
+		return fmt.Errorf("EFI partition not found on install disk %q", installDisk)
+	}
+
+	partitionUUID := efiPartInfo[0].PartitionUUID
+
+	if partitionUUID == nil {
+		return fmt.Errorf("EFI partition UUID not found on install disk %q", installDisk)
+	}
+
+	if err := efivarfs.SetBootEntry(talosBootIndex, &efivarfs.LoadOption{
+		Description: TalosBootEntryDescription,
+		FilePath: efivarfs.DevicePath{
+			&efivarfs.HardDrivePath{
+				PartitionNumber:     1,
+				PartitionStartBlock: efiPartInfo[0].PartitionOffset / uint64(info.SectorSize),
+				PartitionSizeBlocks: efiPartInfo[0].PartitionSize / uint64(info.SectorSize),
+				PartitionMatch: &efivarfs.PartitionGPT{
+					PartitionUUID: *partitionUUID,
+				},
+			},
+			efivarfs.FilePath("/" + sdBootFilePath),
+		},
+	}); err != nil {
+		return fmt.Errorf("failed to set boot entry %d: %w", talosBootIndex, err)
+	}
+
+	currentBootOrder, err := efivarfs.GetBootOrder()
+	if err != nil {
+		return fmt.Errorf("failed to get current BootOrder: %w", err)
+	}
+
+	if currentBootOrder[0] == uint16(talosBootIndex) {
+		// Talos Linux UKI boot entry is already first in the BootOrder
+		return nil
+	}
+
+	if err := efivarfs.SetBootOrder(slices.Concat([]uint16{uint16(talosBootIndex)}, currentBootOrder)); err != nil {
```

Comment:
Not really sure, but could this concatenated slice be longer than the variable? Perhaps that needs checking if I'm not wrong